sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/auth/ServiceAccountIAMTokenProvider.java

```diff
@@ -0,0 +1,154 @@
+package com.mesosphere.sdk.dcos.auth;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.mesosphere.sdk.dcos.http.HttpClientBuilder;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.entity.ContentType;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.time.Instant;
+import java.util.Date;
+
+/**
+ * Provides a token retrieved by `login` operation against IAM service with given service account.
+ *
+ * For reference implementation
+ * @see "https://github.com/mesosphere/bouncer/blob/fc4e0f4205112f3a9bc2b10bb4639d9985beb19e/lynch/lynch/auth.py#L63"
+ */
+public class ServiceAccountIAMTokenProvider implements TokenProvider {
+
+    private URL iamUrl;
+    private String uid;
+    private RSAPrivateKey privateKey;
+    private Executor httpExecutor;
+
+    public ServiceAccountIAMTokenProvider(
```

Comment:
There's a builder, so there's no need (and there really shouldn't be) public constructors.